velox/docs/functions/spark/aggregate.rst

```diff
@@ -11,6 +11,37 @@ General Aggregate Functions
 
     Returns the bitwise XOR of all non-null input values, or null if none.
 
+.. spark:function:: bloom_filter_agg(hash, estimatedNumItems, numBits) -> varbinary
+
+    Creates bloom filter from input hashes and returns it serialized into VARBINARY.
+    The caller is expected to apply xxhash64 function to input data before calling bloom_filter_agg.
+    For example, 
+        bloom_filter_agg(xxhash64(x), 100, 1024)   
+    In Spark implementation, ``estimatedNumItems`` and ``numBits`` are used to decide the number of hash functions and bloom filter capacity.
+    In Velox implementation, ``estimatedNumItems`` is not used.
+
+    ``hash`` cannot be null.
+    ``numBits`` specifies max capacity of the bloom filter, which allows to trade accuracy for memory.
+    In Spark,  the value of``numBits`` is automatically capped at 67,108,864.
+    In Velxo, the value of``numBits`` is automatically capped at 716,800.
+
+    ``x``, ``estimatedNumItems`` and ``numBits`` must be ``BIGINT``.
+
+.. spark:function:: bloom_filter_agg(hash, estimatedNumItems) -> varbinary
+
+    As ``bloom_filter_agg``.
```

Comment:
This description needs to be revised.

```
A version of ``bloom_filter_agg`` that uses ``numBits`` computed as ``estimatedNumItems`` * 8.

``estimatedNumItems`` provides an estimate of the number of values of <TBD: fill in; y seems wrong>.

Value of ``estimatedNumItems`` is capped at 4,000,000.
```

Does 4M cap come from Spark? If so, let's clarify

```
Value of ``estimatedNumItems`` is capped at 4,000,000 like to match Spark's implementation.
```

Reply:
This is Spark implementation https://github.com/apache/spark/blob/branch-3.3/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/aggregate/BloomFilterAggregate.scala#L58

It matches Spark.

Reply:
Now we cannot get configuration when we implement aggregate functions, because we cannot get queryContext config here.
If we need this feature, we should reserve config in `GroupingSet` when we initialize it in `HashAggregation.cpp`.
Now just Spiller::Config exists in `GroupingSet`.
And we need to add a new argument `config` or `context` to functions such as `addRawInput`, it will change all the aggregate function input arguments.
Or we can reserve the config in `Aggregate`, but I don't suggest this way, it change the code less but `Aggregate` is created from FunctionRegistry factory, we cannot use constructor to create it with config.

If you think it is needed, I can help to implement it in another PR.

Reply:
Thank you for pointing this out. Let's create a GitHub issue to explain this use case and discuss how best to implement it. For this PR, let's just mention in the documentation that Spark allows for changing the defaults, but Velox does not.